src/EditorFeatures/CSharpTest/ImplementAbstractClass/ImplementAbstractClassTests_ThroughMember.cs

```diff
@@ -0,0 +1,670 @@
+{
+    Base inner;
+
+    public override event Action Event
+    {
+        add
+        {
+            inner.Event += value;
+        }
+
+        remove
+        {
+            inner.Event -= value;
+        }
+    }
+}", index: 1, title: string.Format(FeaturesResources.Implement_through_0, "inner"));
+        }
+
+        [Fact, WorkItem(41420, "https://github.com/dotnet/roslyn/issues/41420")]
+        public async Task OnlyOverridableMethodsAreOverridden()
+        {
+            await TestInRegularAndScriptAsync(
+@"abstract class Base
+{
+    public abstract void Method();
+
+    public void NonVirtualMethod();
+}
+
+class [|Derived|] : Base
+{
+    Base inner;
+}",
+@"abstract class Base
+{
+    public abstract void Method();
+
+    public void NonVirtualMethod();
+}
+
+class Derived : Base
+{
+    Base inner;
+
+    public override void Method()
+    {
+        inner.Method();
+    }
+}", index: 1, title: string.Format(FeaturesResources.Implement_through_0, "inner"));
+        }
+
+        [Fact, WorkItem(41420, "https://github.com/dotnet/roslyn/issues/41420")]
+        public async Task ProtectedMethodsCannotBeDelegatedThroughBaseType()
+        {
+            await TestExactActionSetOfferedAsync(
+@"abstract class Base
+{
+    protected abstract void Method();
+}
+
+class [|Derived|] : Base
+{
+    Base inner;
+}", new[] { FeaturesResources.Implement_abstract_class });
+        }
+
+        [Fact, WorkItem(41420, "https://github.com/dotnet/roslyn/issues/41420")]
+        public async Task ProtectedMethodsCanBeDelegatedThroughSameType()
+        {
+            await TestInRegularAndScriptAsync(
+@"abstract class Base
+{
+    protected abstract void Method();
+}
+
+class [|Derived|] : Base
+{
+    Derived inner;
+}",
+@"abstract class Base
+{
+    protected abstract void Method();
+}
+
+class Derived : Base
+{
+    Derived inner;
+
+    protected override void Method()
+    {
+        inner.Method();
+    }
+}", index: 1, title: string.Format(FeaturesResources.Implement_through_0, "inner"));
+        }
+
+        [Fact, WorkItem(41420, "https://github.com/dotnet/roslyn/issues/41420")]
+        public async Task ProtectedInternalMethodsAreOverridden()
+        {
+            await TestInRegularAndScriptAsync(
+@"abstract class Base
+{
+    protected internal abstract void Method();
+}
+
+class [|Derived|] : Base
+{
+    Base inner;
+}",
+@"abstract class Base
+{
+    protected internal abstract void Method();
+}
+
+class Derived : Base
+{
+    Base inner;
+
+    protected internal override void Method()
+    {
+        inner.Method();
+    }
+}", index: 1, title: string.Format(FeaturesResources.Implement_through_0, "inner"));
+        }
+
+        [Fact, WorkItem(41420, "https://github.com/dotnet/roslyn/issues/41420")]
+        public async Task InternalMethodsAreOverridden()
+        {
+            await TestInRegularAndScriptAsync(
+@"abstract class Base
+{
+    internal abstract void Method();
+}
+
+class [|Derived|] : Base
```

Comment:
What is the behavior if Derived is defined in a different project that does not have IVT access from project defining Base?

Reply:
it will have an inaccessible member (and will be reported).  i'll add a test.

Reply:
added